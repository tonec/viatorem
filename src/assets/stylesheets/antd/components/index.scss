@import "./affix/index";
@import "./alert/index";
@import "./anchor/index";
@import "./auto-complete/index";
@import "./avatar/index";
@import "./back-top/index";
@import "./badge/index";
@import "./breadcrumb/index";
@import "./button/index";
@import "./calendar/index";
@import "./card/index";
@import "./carousel/index";
@import "./cascader/index";
@import "./checkbox/index";
@import "./collapse/index";
@import "./date-picker/index";
@import "./divider/index";
@import "./dropdown/index";
@import "./form/index";
@import "./grid/index";
@import "./input/index";
@import "./input-number/index";
@import "./layout/index";
@import "./list/index";
@import "./mention/index";
@import "./menu/index";
@import "./message/index";
@import "./modal/index";
@import "./notification/index";
@import "./pagination/index";
@import "./popover/index";
@import "./progress/index";
@import "./radio/index";
@import "./rate/index";
@import "./select/index";
@import "./slider/index";
@import "./spin/index";
@import "./steps/index";
@import "./switch/index";
@import "./table/index";
@import "./tabs/index";
@import "./tag/index";
@import "./time-picker/index";
@import "./timeline/index";
@import "./tooltip/index";
@import "./transfer/index";
@import "./tree/index";
@import "./tree-select/index";
@import "./upload/index";