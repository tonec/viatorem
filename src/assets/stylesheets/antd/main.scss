@import "./themes/default";
@import "./core/index";
@import "./components/index";

div#root {
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  background: #F0F2F5;

  > div {
    width: 100%;
  }
}
