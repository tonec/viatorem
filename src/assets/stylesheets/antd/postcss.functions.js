const tinyColor = require('./functions/tinyColor.js')
const bezierEasing = require('./functions/bezierEasing.js')
const colorPalette = require('./functions/colorPalette.js')

module.exports = { tinyColor, bezierEasing, colorPalette }
