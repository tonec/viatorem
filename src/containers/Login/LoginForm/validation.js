import memoize from 'lru-memoize'
import { createValidator, required, email } from 'utils/validation'

const validation = createValidator({
  email: [email, required],
  password: required
})

export default memoize(10)(validation)
