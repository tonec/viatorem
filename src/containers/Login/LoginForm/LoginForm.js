import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Field } from 'redux-form'
import { Form, Button, Alert } from 'antd'
import { TextField } from 'components/Forms/Inputs'

import style from './style.scss'

export default class LoginForm extends Component {

  static propTypes = {
    handleSubmit: PropTypes.func.isRequired,
    onSubmit: PropTypes.func.isRequired,
    submitting: PropTypes.bool,
    error: PropTypes.any
  }

  render () {
    const { handleSubmit, onSubmit, submitting, error } = this.props

    return (
      <Form layout="vertical" onSubmit={handleSubmit(onSubmit)}>
        <Field
          name="email"
          type="text"
          size="large"
          component={TextField}
          placeholder="Email"
        />
        <Field
          name="password"
          type="password"
          size="large"
          component={TextField}
          placeholder="Password"
        />
        <Form.Item>
          <Button
            type="primary"
            size="large"
            htmlType="submit"
            loading={submitting}
            className={style.button}
          >
            Log In
          </Button>
        </Form.Item>
        {error && <Alert showIcon type="error" message={error} />}
      </Form>
    )
  }
}
