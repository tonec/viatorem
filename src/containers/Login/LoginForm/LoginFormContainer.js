import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { compose } from 'redux'
import { connect } from 'react-redux'
import { reduxForm } from 'redux-form'
import { login } from 'redux/modules/auth/actions'
import validation from './validation'
import LoginForm from './LoginForm'

export class LoginFormContainer extends Component {

  static propTypes = {
    login: PropTypes.func.isRequired
  }

  render () {
    return (
      <LoginForm {...this.props} onSubmit={this.props.login} />
    )
  }
}

const mapState = () => ({})

export default compose(
  connect(mapState, { login }),
  reduxForm({
    form: 'login',
    validate: validation
  })
)(LoginFormContainer)
