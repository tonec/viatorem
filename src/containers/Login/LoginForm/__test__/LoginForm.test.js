import React from 'react'
import { shallow } from 'enzyme'
import { Alert } from 'antd'
import LoginForm from '../LoginForm'

describe('LoginForm', () => {

  const props = {
    handleSubmit: jest.fn(),
    onSubmit: jest.fn(),
    submitting: false,
    error: null
  }

  it('renders according to snapshot', () => {
    const component = shallow(<LoginForm {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('display the error response when submission fails', () => {
    const error = 'Authentication failed'
    const component = shallow(<LoginForm {...props} error={error} />)
    expect(component.find(Alert)).toHaveLength(1)
  })
})
