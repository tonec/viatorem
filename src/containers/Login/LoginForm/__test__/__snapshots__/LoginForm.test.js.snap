// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders according to snapshot 1`] = `
<Form
  hideRequiredMark={false}
  layout="vertical"
  onSubmit={[Function]}
  prefixCls="ant-form"
>
  <Field
    component={[Function]}
    name="email"
    placeholder="Email"
    size="large"
    type="text"
  />
  <Field
    component={[Function]}
    name="password"
    placeholder="Password"
    size="large"
    type="password"
  />
  <FormItem
    colon={true}
    hasFeedback={false}
    prefixCls="ant-form"
  >
    <Button
      ghost={false}
      htmlType="submit"
      loading={false}
      prefixCls="ant-btn"
      size="large"
      type="primary"
    >
      Log In
    </Button>
  </FormItem>
</Form>
`;
